workflow/controller/controller.go

```diff
@@ -227,12 +229,21 @@ func (wfc *WorkflowController) processNextItem() bool {
 	}
 
 	if wf.ObjectMeta.Labels[common.LabelKeyCompleted] == "true" {
+		wfc.throttler.NotifyProcessed(key)
 		// can get here if we already added the completed=true label,
 		// but we are still draining the controller's workflow workqueue
 		return true
 	}
-	woc := newWorkflowOperationCtx(wf, wfc)
-	woc.operate()
+
+	if wfc.throttler.CanProcess(key, wf.Spec.Priority) {
+		woc := newWorkflowOperationCtx(wf, wfc)
+		woc.operate()
```

Comment:
I think the logic should be that any workflow which has already started, should be allowed to continue to operate. So priority only takes affect during admission, but once the workflow is admitted, it should be allowed to complete. This code appears to halt all processing of any workflow (even if already running), so long as theres a higher one in the queue.

Reply:
The in-progress workflows are not halted. Throttler allows executing workflow if it is already in progress:
```
func (t *throttler) CanProcess(key interface{}, priority int) bool {
    _, isInProgress := t.processingKeys[key]
    shouldProcess := isInProgress || *t.parallelism > len(t.processingKeys)
```

Reply:
Ok I missed that part. Thanks.